src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/Interop/LdapPal.Linux.cs

```diff
@@ -177,6 +177,25 @@ internal static int StartTls(ConnectionHandle ldapHandle, ref int serverReturnVa
 
         internal static IntPtr StringToPtr(string s) => Marshal.StringToHGlobalAnsi(s);
 
+        internal static unsafe IntPtr StringToPtr(string s, out int length)
+        {
+            var pointer = StringToPtr(s);
+            length = 0;
+
+            if (pointer == IntPtr.Zero)
+            {
+                return pointer;
+            }
+
+            byte* stringPtr = (byte*)pointer;
+            while (stringPtr[length] != '\0')
+            {
+                length++;
+            }
```

Comment:
If `StringToPtr` is not used elsewhere, you can shorten the method to this. Either way a `while` loop to find the NUL is not advised because it won't be vectorized.

```suggestion
        internal static unsafe IntPtr StringToPtr(string s, out int length)
        {
            var pointer = Marshal.StringToHGlobalAnsi(s);
            length = MemoryMarshal.CreateReadOnlySpanFromNullTerminated(pointer).Length;
```

Reply:
Thank you! I didn't know about this before. Applied this suggestion partly as a new commit.
Also, StringToPtr is used in other methods but they don't use string length. So that's why I added this overload with `out int length`

Reply:
Another thing I thought is to not add this new overload and instead calculate the length at the callsite where it's needed.

Reply:
Something like that? https://github.com/Sparin/runtime/commit/78484539924277e2295756616a639d1d55c14f95
Tell me if I should apply this commit to PR

Reply:
Yeah. It's up to you if you like it more.

Reply:
Done. Seems cleaner now for me than the first one :D